code/modules/mob/living/carbon/xenomorph/abilities.dm

```diff
@@ -597,134 +597,40 @@
 		return FALSE
 	if(!owner.Adjacent(A))
 		if(!silent)
-			to_chat(owner, span_warning("\The [A] is too far away."))
+			owner.balloon_alert(owner, "[A] is too far away")
 		return FALSE
-	if(isobj(A))
-		var/obj/O = A
-		if(CHECK_BITFIELD(O.resistance_flags, RESIST_ALL))
-			if(!silent)
-				to_chat(owner, span_warning("We cannot dissolve \the [O]."))
-			return FALSE
-		if(O.acid_check(acid_type))
-			if(!silent)
-				to_chat(owner, span_warning("This object is already subject to a more or equally powerful acid."))
-			return FALSE
-		if(istype(O, /obj/structure/window_frame))
-			var/obj/structure/window_frame/WF = O
-			if(WF.reinforced && acid_type != /obj/effect/xenomorph/acid/strong)
-				if(!silent)
-					to_chat(owner, span_warning("This [WF.name] is too tough to be melted by our weak acid."))
-				return FALSE
-	else if(isturf(A))
-		var/turf/T = A
-		if(T.acid_check(acid_type))
-			if(!silent)
-				to_chat(owner, span_warning("This object is already subject to a more or equally powerful acid."))
-			return FALSE
-		if(iswallturf(T))
-			var/turf/closed/wall/wall_target = T
-			if(wall_target.acided_hole)
-				if(!silent)
-					to_chat(owner, span_warning("[wall_target] is already weakened."))
-				return FALSE
-
-/obj/proc/acid_check(obj/effect/xenomorph/acid/new_acid)
-	if(!new_acid)
-		return TRUE
-	if(!current_acid)
-		return FALSE
-
-	if(initial(new_acid.acid_strength) > current_acid.acid_strength)
+	if(A.resistance_flags & UNACIDABLE || !A.dissolvability(initial(acid_type.acid_strength)))
+		if(!silent)
+			owner.balloon_alert(owner, "We cannot dissolve [A]")
 		return FALSE
-	return TRUE
-
-/turf/proc/acid_check(obj/effect/xenomorph/acid/new_acid)
-	if(!new_acid)
-		return TRUE
-	if(!current_acid)
+	if(!A.should_apply_acid(initial(acid_type.acid_strength)) || initial(acid_type.acid_strength) == initial(A.current_acid?.acid_strength))
+		if(!silent)
+			owner.balloon_alert(owner, "[A] is already subject to a more or equally powerful acid")
 		return FALSE
-
-	if(initial(new_acid.acid_strength) > current_acid.acid_strength)
+	if(ismob(A))
+		if(!silent)
+			owner.balloon_alert(owner, "We can't melt [A]")
```

Comment:
put the mob check before the acid strength stuff, since theres no point in running that proc if its a mob anyway.